hooks/useRefreshCandyMachine.tsx

```diff
@@ -0,0 +1,87 @@
+import { useAnchorWallet, useWallet } from '@solana/wallet-adapter-react'
+import { AnchorProvider, Program, BN } from '@project-serum/anchor'
+import { useRPC } from 'hooks'
+import { PublicKey, LAMPORTS_PER_SOL, Connection } from '@solana/web3.js'
+import { useState } from 'react'
+import { CANDY_MACHINE_PROGRAM_V2_ID } from 'lib/candy-machine/mint/constants'
+import { CandyMachineAccount } from 'lib/candy-machine/interfaces'
+
+const useRefreshCandyMachine = (account: string) => {
+    const anchorWallet = useAnchorWallet()
+    const { connection, network } = useRPC()
+    const [itemsRemaining, setItemsRemaining] = useState(0)
+    const [itemsAvailable, setItemsAvailable] = useState(0)
+    const [nftPrice, setNftPrice] = useState(0)
+    const [isActive, setIsActive] = useState(false)
+    const [isCaptcha, setIsCaptcha] = useState(false)
+    const [candyMachine, setCandyMachine] = useState<CandyMachineAccount>()
+
+    const wallet = useWallet()
+
+    async function refreshCandyMachineState() {
+        try {
+            if (!anchorWallet || !wallet.publicKey || !connection || !network) return
+            const provider = new AnchorProvider(connection, anchorWallet, {
+                preflightCommitment: 'recent',
+            })
+
+            const idl = await Program.fetchIdl(CANDY_MACHINE_PROGRAM_V2_ID, provider)
+
+            const program = new Program(idl!, CANDY_MACHINE_PROGRAM_V2_ID, provider)
```

Comment:
Instead of adding the ! to the idl, on the line above we could check if there is an idl, else throw an error.